manta-crypto/src/ecc.rs

```diff
@@ -334,6 +334,36 @@ where
     }
 }
 
+// TODO： Simplify the trait properties here.
+// 1. We need `G::Scalar: Clone` and `Self::Key: Clone` since `mul()` takes by value instead of by reference.
+// 2. We need `G: ScalarMul<COM, Output=G>` since the output will be used for `scalar_mul` again.
+impl<G, COM> RandomizableKeyDerivationFunction<COM> for DiffieHellman<G>
+where
+    G: ScalarMul<COM, Output = G>,
+    G::Scalar: Clone + Mul<COM>,
+    Self::Key: Clone,
+{
+    type Randomness = G::Scalar;
+
+    fn rand_input_in(
+        &self,
+        randomness: &Self::Randomness,
+        input: &Self::Key,
+        compiler: &mut COM,
+    ) -> Self::Key {
+        G::Scalar::mul(randomness.clone(), input.clone(), compiler)
+    }
+
+    fn rand_output_in(
+        &self,
+        randomness: &Self::Randomness,
+        output: &Self::Output,
+        compiler: &mut COM,
+    ) -> Self::Output {
+        output.scalar_mul(randomness, compiler)
+    }
+}
+
```

Comment:
TODO:
1. Add a test on the `Random Derivation` property.
2. The [test for key derivation function](https://github.com/Manta-Network/manta-rs/pull/82/files#diff-0a2f47f9702f7f522d33e2efcc120d9b919caf59c7ca309f8f04719f57cc3c7bR619-R629) is never called. We may need to also add a test for it.